pkg/semtech-loramac/contrib/semtech_loramac_getset.c

```diff
@@ -27,217 +29,303 @@
+{
+    return mac->port;
+}
+
+void semtech_loramac_set_tx_mode(semtech_loramac_t *mac, uint8_t mode)
+{
+    mac->cnf = mode;
 }
 
-void semtech_loramac_set_rx2_freq(uint8_t freq)
+uint8_t semtech_loramac_get_tx_mode(semtech_loramac_t *mac)
 {
-    Rx2ChannelParams_t params;
+    return mac->cnf;
+}
+
+void semtech_loramac_set_rx2_freq(semtech_loramac_t *mac, uint32_t freq)
+{
+    mutex_lock(&mac->lock);
+    DEBUG("[semtech-loramac] setting RX2 freq to %lu\n", freq);
+    Rx2ChannelParams_t p;
     MibRequestConfirm_t mibReq;
     mibReq.Type = MIB_RX2_DEFAULT_CHANNEL;
     LoRaMacMibGetRequestConfirm(&mibReq);
-    params.Frequency = freq;
-    params.Datarate = mibReq.Param.Rx2DefaultChannel.Datarate;
-    _loramac_set_rx2_params(params.Frequency, params.Datarate);
+    p.Frequency = freq;
+    p.Datarate = mibReq.Param.Rx2DefaultChannel.Datarate;
+    semtech_loramac_channel_params_t params;
+    params.frequency = freq;
+    params.datarate = p.Datarate;
+    semtech_loramac_set_rx2_params(mac, params);
+    mutex_unlock(&mac->lock);
 }
 
-uint32_t semtech_loramac_get_rx2_freq(void)
+uint32_t semtech_loramac_get_rx2_freq(semtech_loramac_t *mac)
 {
+    mutex_lock(&mac->lock);
+    uint32_t freq;
+    DEBUG("[semtech-loramac] getting RX2 freq\n");
     MibRequestConfirm_t mibReq;
     mibReq.Type = MIB_RX2_DEFAULT_CHANNEL;
     LoRaMacMibGetRequestConfirm(&mibReq);
-    return mibReq.Param.Rx2DefaultChannel.Frequency;
+    freq = mibReq.Param.Rx2DefaultChannel.Frequency;
+    mutex_unlock(&mac->lock);
+    return freq;
 }
 
-void semtech_loramac_set_rx2_dr(uint8_t dr)
+void semtech_loramac_set_rx2_dr(semtech_loramac_t *mac, uint8_t dr)
 {
-    Rx2ChannelParams_t params;
+    mutex_lock(&mac->lock);
+    DEBUG("[semtech-loramac] setting RX2 datarate to %d\n", dr);
+    Rx2ChannelParams_t p;
     MibRequestConfirm_t mibReq;
     mibReq.Type = MIB_RX2_DEFAULT_CHANNEL;
     LoRaMacMibGetRequestConfirm(&mibReq);
-    params.Datarate = dr;
-    params.Frequency = mibReq.Param.Rx2DefaultChannel.Frequency;
-    _loramac_set_rx2_params(params.Frequency, params.Datarate);
+    p.Datarate = dr;
+    p.Frequency = mibReq.Param.Rx2DefaultChannel.Frequency;
+    semtech_loramac_channel_params_t params;
+    params.datarate = dr;
+    params.frequency = p.Frequency;
+    semtech_loramac_set_rx2_params(mac, params);
+    mutex_unlock(&mac->lock);
 }
 
-uint8_t semtech_loramac_get_rx2_dr(void)
+uint8_t semtech_loramac_get_rx2_dr(semtech_loramac_t *mac)
 {
+    mutex_lock(&mac->lock);
+    uint8_t datarate;
+    DEBUG("[semtech-loramac] getting RX2 datarate\n");
     MibRequestConfirm_t mibReq;
     mibReq.Type = MIB_RX2_DEFAULT_CHANNEL;
     LoRaMacMibGetRequestConfirm(&mibReq);
-    return mibReq.Param.Rx2DefaultChannel.Datarate;
+    datarate = mibReq.Param.Rx2DefaultChannel.Datarate;
+    mutex_unlock(&mac->lock);
+    return datarate;
+}
+
+void semtech_loramac_set_rx2_params(semtech_loramac_t *mac,
```

Comment:
I have a compilation error here because `semtech_loramac_t *mac` argument is not used.

Reply:
Fixed